### ISSUE TITLE
CP & IV per pokemon filter ignored if moveset is empty

### ISSUE THREAD

**Issue by N0ct3** [NECROBOTIO/NecroBot#4811]
There is currently a big issue in the **GetDuplicatePokemonToTransfer** method:
while trying to determine which pokemon has to be transfered, the method is iterating over the provided moveset before applying IV and/or CP filter: 
```
pokemonTransferFilter.Moves.ForEach(moveset =>
                            {
                                ret = !pokemonTransferFilter.MovesOperator.BoolFunc(
                                    moveset.Intersect(new[] { p.Move1, p.Move2 }).Any(),
                                    pokemonTransferFilter.KeepMinOperator.BoolFunc(
                                        p.Cp >= pokemonTransferFilter.KeepMinCp,
                                        PokemonInfo.CalculatePokemonPerfection(p) >= pokemonTransferFilter.KeepMinIvPercentage,
                                        pokemonTransferFilter.KeepMinOperator.ReverseBoolFunc(
                                            pokemonTransferFilter.KeepMinOperator.InverseBool(pokemonTransferFilter.UseKeepMinLvl),
                                            PokemonInfo.GetLevel(p) >= pokemonTransferFilter.KeepMinLvl)));
                            });
```
Except that if we did not provide any move, the CP & IV filter will simply be ignored and the pokemon selected for transfer...
This change was introduced with changeset  e480da0e

**Comment by theloser** [comment]
@N0ct3 Please check pull request #4705 -- seems like someone already fixed it, we just have to wait unitl it is merged into master.

**Comment by N0ct3** [comment]
Thanks for the update!
Luckily I stopped my bot before all my pokemons were gone!

**Comment by Naprosnia** [comment]
I got:
"PrioritizeIvOverCp": true
"KeepMinIvPercentage": 100.0
"KeepMinDuplicatePokemon": 1

and got 2 Eevee with iv 100, when i run this new version, one of them was tranfered.
This problem is related to this error?

**Comment by theloser** [comment]
@Naprosnia Very much, yes.

**Comment by actislav** [comment]
fixed better in #4827
